### PR TITLE
11305 validate only two legends per layer

### PR DESCRIPTION
**Pull request by javitonino**
More validations #11305 

**Acceptance**
Create, update, delete some layers. Everything should work. Keep an eye on the network console, you should not get errors from backend.

### CODE REVIEW

File: app/models/carto/legend.rb [hunk 1]
```diff
@@ -57,6 +62,16 @@ def under_max_legends_per_layer
       end
     end
 
+    def under_max_legends_per_layer_and_type
+      return unless type
+      LEGEND_TYPES_PER_ATTRIBUTE.each do |attribute, legend_types|
+        next unless legend_types.include?(type)
+
+        other_legends_present = layer.legends.none? { |legend| legend.id != id && legend_types.include?(legend.type) }
```
Comment (juanignaciosl) [inline app/models/carto/legend.rb]: How come `other_legends_present`, `none?` plus `unless`?  Isn't this actually `any?` + `if`?

File: NEWS.md [hunk 2]
```diff
@@ -63,6 +63,7 @@ Development
 * Avoid loading all rake code in resque workers (#11069)
 * Fix analysis notification in running state (#11079)
 * Fix color for "Other" category (#11078)
+* Validate that lwe only create one legend per type (#11556)
```
Comment (juanignaciosl) [inline NEWS.md]: Typo.

File: app/models/carto/legend.rb (hunk 1 above)
Comment (javitonino) [inline app/models/carto/legend.rb]: Oh yeah, much clearer with `if` + `any?`. Plus, the variable name didn't help at all (what was I thinking? 😅)

**Comment by juanignaciosl** [comment]
Cool, when I first read it my head exploded :D

**Pull request merged**